lib/jekyll/commands/doctor.rb

```diff
@@ -41,9 +41,20 @@ def healthy?(site)
             !conflicting_urls(site),
             !urls_only_differ_by_case(site),
             proper_site_url?(site),
+            properly_gathered_posts?(site),
           ].all?
         end
 
+        def properly_gathered_posts?(site)
+          return true if site.config["collections_dir"].empty?
+          posts_at_root = site.in_source_dir("_posts")
+          return true unless File.directory?(posts_at_root)
+          Jekyll.logger.warn "Warning:", "Contents of #{posts_at_root} will not be " \
+                             "processed since you have specified a custom directory " \
+                             "to house all collections."
```

Comment:
Could we rater display the awaited action from the user?

Warning: Custom `collection_dir` detected : move the `_posts` directory inside  the `my_collections` directory.